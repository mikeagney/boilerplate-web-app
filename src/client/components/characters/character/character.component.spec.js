import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ClickableEdit from '../../controls/clickable-edit';
import Character from './character.component';

describe('Character component', () => {
  const defaultProps = {
    characterId: 'b2b',
    name: 'Mock name',
    setName: jest.fn(),
  };

  it('will render Character component', () => {
    // Arrange
    const props = {
      ...defaultProps,
    };

    // Act
    const wrapper = shallow(<Character {...props} />);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('will call setText when ClickableEdit sets text', () => {
    // Arrange
    const props = {
      ...defaultProps,
    };
    const wrapper = shallow(<Character {...props} />);
    const { setText } = wrapper.find(ClickableEdit).props();

    // Act
    setText('new text');

    // Assert
    expect(props.setName).toHaveBeenCalledWith(props.characterId, 'new text');
  });
});
