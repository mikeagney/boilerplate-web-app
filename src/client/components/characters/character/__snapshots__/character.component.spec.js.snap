// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component will render Character component 1`] = `
<Card
  as="div"
  body={false}
  className="character"
  key="b2b"
>
  <CardHeader
    as="h5"
    className="bg-primary text-white"
  >
    <ClickableEdit
      setText={[Function]}
      text="Mock name"
    />
  </CardHeader>
  <CardBody>
    <CardText>
      Skills go here
    </CardText>
  </CardBody>
</Card>
`;
